### ISSUE TITLE
Generic handling of unknown types in PostgreSQL sources

### ISSUE THREAD

**Issue by chaas** [MaterializeInc/materialize#15073]
When customers create a source (e.g. from postgres), if we don't support any of the column types, it fails to create. We can solve this issues one-by-one by adding support of each postgres type as the request comes in, but the turnaround time & interruption load on the team makes that unsustainable and undesirable. In addition, some customers may use postgres extensions which have additional types we may not choose to support.

While we ultimately want to add native support for all (or at least most) postgres types long-term, we need an interim solution that unblocks all customers now.

One proposed solution as discussed in [#14016](https://github.com/MaterializeInc/materialize/issues/14016#issuecomment-1208299873) is to re-cast unsupported types to string.
e.g. `CAST foo.my_enum AS string`.

Per suggestion from @mjibson, we could send NOTICE messages so it's not completely invisible to the user.

An example of a customer experiencing this pain point: #14748

Some workarounds/alternative solutions:
* #14749 (we should support this regardless)
* Selected column replication set up by the user in postgres (only available in postgres 15+)

- [x] #15502

**Comment by chaas** [comment]
cc: @nmeagan11

**Comment by benesch** [comment]
@chaas can we retitle to "Generic support for handling unknown types in PostgreSQL sources", or something like that? Whatever we come up with for PostgreSQL sources is going to be PostgreSQL specific, for example, and won't generalize to e.g. Kafka sources.

**Comment by benesch** [comment]
We should consider improving the "type with OID ..." error message in the meantime. It could do two things better:

  * Add a hint indicating that this is likely to be an unsupported type in Materialize.
  * Indicate which column of which table uses the unsupported type.

**Comment by nmeagan11** [comment]
Linking https://github.com/MaterializeInc/materialize/issues/15716 where active conversation is happening.

**Comment by benesch** [comment]
Actually let's just close this as a retroactive duplicate of #15716!
